vllm/spec_decode/multi_head_worker.py

```diff
@@ -0,0 +1,63 @@
+import weakref
+from typing import List, Tuple
+
+import torch
+
+from vllm.sequence import ExecuteModelRequest, SamplerOutput
+from vllm.spec_decode.interfaces import SpeculativeProposals
+from vllm.spec_decode.top1_proposer import Top1Proposer
+from vllm.worker.worker import Worker
+
+
+class MultiHeadWorker(Worker):
+    """The MultiHeadWorker is equivalent to a Worker except that it allows
+    multiple forward passes in a single call, assuming the scheduler has
+    allocated enough space to store the additional KV. This reduces overhead
+    by invoking the scheduler less.
```

Comment:
Do we need to inherit from the worker?

Reply:
We need to use the `init_device` method from worker. Removing inheritance from worker might require duplication of that method. Other than that, I don't think there's any dependence.